### DISCUSSION TITLE
momentum

### DISCUSSION THREAD

**Discussion by motahareh89** [epochpic/epoch#627] [General]
hello! I want plot electron velocity but our datas are from order 10^-22 is it ok?

**Comment by Status-Mirror** [comment]
Hey @motahareh89,

Sorry for the delay. It sounds like you're looking at particle momenta. EPOCH uses SI units, so these are in kgm/s. A 1e-22 kgm/s electron has a velocity of:

$p = \gamma m_e v$
$E = \sqrt{p^2 c^2 + m_e^2 c^4} = \gamma m_e c^2$
$v = p c^2 / E$

hence, for $p = 10^{-22}$ kgm/s, $v \approx 1.0\times 10^8$ m/s.

Hope this helps,
Stuart

**Comment by motahareh89** [comment]
Dear Sturt,
I hope this email finds you well,
Thank you for your guidance, I have used epoch recently for my research so
may I ask if I have a question directory by this Email?
warm regards,
Motahareh

**Reply by Status-Mirror** [reply]
Hey Motahareh,

For many cases, you might find our [documentation webpage](https://epochpic.github.io/) a useful resource.

If the documentation isn't clear on something, or you're still having problems, feel free to check in on these GitHub pages - we'll generally try to provide support if we can.

Cheers,
Stuart

**Comment by motahareh89** [comment]
Thank you for responding.

**Comment by motahareh89** [comment]
Dear Dr. Morris,
 I hope you are doing well.
I'm using EPOCH for THz radiation.
I have two beams with the properties that I wrote in the code. They are
beating in plasma and then THz should be generated. But my result is vague
Here is my code.
Please help me, I have to prepare my proposal.
regards,
Motahareh
